### PR TITLE
Fix docs: splitting can introduce a hyphen

### PR DESCRIPTION
**Pull request by Minoru**
Just a small typo I found while reviewing this for cargo-crev.

### CODE REVIEW

File: src/splitting.rs [hunk 1]
```diff
@@ -24,9 +24,9 @@
 /// [`hyphenation` documentation]: https://docs.rs/hyphenation/
 pub trait WordSplitter {
     /// Return all possible splits of word. Each split is a triple
-    /// with a head, a hyphen, and a tail where `head + &hyphen +
-    /// &tail == word`. The hyphen can be empty if there is already a
-    /// hyphen in the head.
+    /// with a head, a hyphen, and a tail where `head + &tail == word`.
```
Comment (mgeisler) [inline src/splitting.rs:27]: The return type is `Vec<(&'w str, &'w str, &'w str)>` and I was trying to describe how the elements of the 3-tuples here should combine back to the original word. So I think the `hyphen` part of the docstring is still important. Am I missing something?

**Comment by mgeisler** [comment]
Hi @Minoru, thanks for the PR. Can you explain more why you want to remove `hyphen` from the docstring?

File: src/splitting.rs (hunk 1 above)
Comment (Minoru) [reply src/splitting.rs:27]: Splitting a word can introduce a hyphen that wasn't in the original word. In that case, the three components won't combine into the input word anymore.

Taking the example from the docstring: the input is "technology" (no hyphens), but only the last result yields this string if all three components are concatenated:

```
("tech", "-", "nology"), // "tech" + "-" + "nology" == "tech-nology" != "technology"
("technol", "-", "ogy"), // "technol-ogy" != "technology"
("technolo", "-", "gy"), // "technolo-gy" != "technology"
("technology", "", "")];
```

So I fixed the relation in the docstring to exclude `hyphen`.

Sorry I didn't include a rationale when I filed the PR. I should've learned by now that the simpler the change, the less obvious the motivation is, so I should explain it in more detail.

**Comment by Minoru** [comment]
AppVeyor failure is not related to my change, BTW.

File: src/splitting.rs (hunk 1 above)
Comment (mgeisler) [reply src/splitting.rs:27]: Oh my! You're completely right 🤦‍♂️ It had never occurred to me until now that the pieces _don't_ combine to the original word in most cases!

**Pull request merged**